### PR TITLE
IN county tax

### PR DESCRIPTION
**Pull request by MattHJensen**

### CODE REVIEW

**Comment by MattHJensen** [comment]
@nikhilwoodruff, @MaxGhenis could you take a look at this implementation and let me know if I'm on the right track or would prefer that I head a different direction?

File: openfisca_us/parameters/gov/states/in/tax/income/taxes/county/rates.yaml [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  period: year
+  unit: /1
+  name: in_county_tax_rates
+  label: IN county tax rates
+  reference:
+  - title: IC 6-3.6 ARTICLE 3.6. LOCAL INCOME TAXES
+    href: http://iga.in.gov/legislative/laws/2021/ic/titles/006#6-3.6
+
+  breakdown:
+    - county_enum
```
Comment (nikhilwoodruff) [inline openfisca_us/parameters/gov/states/in/tax/income/taxes/county/rates.yaml:12]: Could you remove the breakdown? The state-only performance tricks mean we don't have to use it, and it'd increase performance by not having to add a zero rate for every non-IN county.

File: openfisca_us/parameters/gov/states/in/tax/income/taxes/county/rates.yaml [hunk 2]
```diff
@@ -0,0 +1,197 @@
+description: IN county tax rates.
+metadata:
+  period: year
+  unit: /1
+  name: in_county_tax_rates
+  label: IN county tax rates
+  reference:
+  - title: IC 6-3.6 ARTICLE 3.6. LOCAL INCOME TAXES
+    href: http://iga.in.gov/legislative/laws/2021/ic/titles/006#6-3.6
+
```
Comment (nikhilwoodruff) [inline openfisca_us/parameters/gov/states/in/tax/income/taxes/county/rates.yaml:10]:
```suggestion
  propagate_metadata_to_children: true
```

File: openfisca_us/tests/policy/baseline/gov/states/in/tax/income/taxes/in_county_tax.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+- name: IN county tax
+  period: 2021
+  input:
+    Households:
+      household1:
+        county_enum: MONROE_COUNTY_IN
```
Comment (nikhilwoodruff) [inline openfisca_us/tests/policy/baseline/gov/states/in/tax/income/taxes/in_county_tax.yaml:6]:
```suggestion
        county: MONROE_COUNTY_IN
```

File: openfisca_us/tests/policy/baseline/gov/states/in/tax/income/taxes/in_county_tax.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+- name: IN county tax
+  period: 2021
+  input:
+    Households:
```
Comment (nikhilwoodruff) [inline openfisca_us/tests/policy/baseline/gov/states/in/tax/income/taxes/in_county_tax.yaml:4]:
```suggestion
    households:
```

File: openfisca_us/tests/policy/baseline/gov/states/in/tax/income/taxes/in_county_tax.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+      household1:
+        county_enum: MONROE_COUNTY_IN
+    tax_units:
+      tax_unit:
+        members: household1
```
Comment (nikhilwoodruff) [inline openfisca_us/tests/policy/baseline/gov/states/in/tax/income/taxes/in_county_tax.yaml:9]: Members can only be people, not households

File: openfisca_us/variables/gov/states/in/tax/income/taxes/in_county_tax.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    label = "IN county tax"
+    definition_period = YEAR
+    unit = USD
+    reference = "http://iga.in.gov/legislative/laws/2021/ic/titles/006#6-3-2-1"  # (a)(3)
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states["in"].tax.income.taxes.county
+        in_agi = tax_unit("in_agi", period)
+        # county calculations are at the person level for each taxpayer in the law
+        county = tax_unit.household("county_enum", period)
```
Comment (nikhilwoodruff) [inline openfisca_us/variables/gov/states/in/tax/income/taxes/in_county_tax.py:16]:
```suggestion
        county = tax_unit.household("county_str", period)
```

File: openfisca_us/variables/gov/states/in/tax/income/taxes/in_county_tax.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class in_county_tax(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "IN county tax"
+    definition_period = YEAR
+    unit = USD
+    reference = "http://iga.in.gov/legislative/laws/2021/ic/titles/006#6-3-2-1"  # (a)(3)
+
```
Comment (nikhilwoodruff) [inline openfisca_us/variables/gov/states/in/tax/income/taxes/in_county_tax.py:11]:
```suggestion
defined_for = StateCode.IN
```

**Review by nikhilwoodruff — COMMENTED** [review]
Thanks @MattHJensen - looks good so far, this will fit nicely with the MD county tax structure.

**Comment by MattHJensen** [comment]
@nikhilwoodruff, thanks for the help above. I could use another tip on the versioning failure, and otherwise this is ready for review.

**Comment by nikhilwoodruff** [comment]
> @nikhilwoodruff, thanks for the help above. I could use another tip on the versioning failure (also happening in #1168), and otherwise this is ready for review. cc @MaxGhenis

No problem - could you try pulling the master branch into this one? I just did that locally and there were some changes to the changelog that I think are needed (didn't want to mess up your local version).

**Comment by MattHJensen** [comment]
> No problem - could you try pulling the master branch into this one? I just did that locally and there were some changes to the changelog that I think are needed (didn't want to mess up your local version).

That makes a lot of sense and of course worked. Thanks.

**Comment by MattHJensen** [comment]
@nikhilwoodruff and @MaxGhenis, this one is ready for review now. Thanks.

File: openfisca_us/tests/policy/baseline/gov/states/in/tax/income/taxes/in_county_tax.yaml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+- name: IN county tax
+  period: 2021
+  input:
+    people:
```
Comment (MaxGhenis) [inline openfisca_us/tests/policy/baseline/gov/states/in/tax/income/taxes/in_county_tax.yaml:4]: We reserve `variable_name.yaml` for unit testing `variable_name`, so the only inputs are the inputs to the function itself. More involved tests would go in an `integration.yaml` in the same folder if you'd like to add that. Though in this case I think the unit test suffices.
```suggestion
    in_agi: 1_000
    county_str: MONROE_COUNTY_IN
```

File: openfisca_us/variables/gov/states/in/tax/income/taxes/in_county_tax.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    label = "IN county tax"
+    definition_period = YEAR
+    unit = USD
+    reference = "http://iga.in.gov/legislative/laws/2021/ic/titles/006#6-3-2-1"  # (a)(3)
+    defined_for = StateCode.IN
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states["in"].tax.income.taxes.county
+        in_agi = tax_unit("in_agi", period)
+        # county calculations are at the person level for each taxpayer in the law
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/states/in/tax/income/taxes/in_county_tax.py:16]: Does this comment refer to the `in_agi` calculation?
```suggestion
        # County calculations are at the person level for each taxpayer in the law.
```

File: openfisca_us/parameters/gov/states/in/tax/income/taxes/county/rates.yaml [hunk 10]
```diff
@@ -0,0 +1,195 @@
+description: IN county tax rates.
+metadata:
+  period: year
+  unit: /1
+  name: in_county_tax_rates
+  label: IN county tax rates
+  reference:
+  - title: IC 6-3.6 ARTICLE 3.6. LOCAL INCOME TAXES
+    href: http://iga.in.gov/legislative/laws/2021/ic/titles/006#6-3.6
+  propagate_metadata_to_children: true
```
Comment (MaxGhenis) [inline openfisca_us/parameters/gov/states/in/tax/income/taxes/county/rates.yaml:10]: I don't think we need this here.

**Comment by MaxGhenis** [comment]
@PavelMakarchuk could you get this across the finish line? A la https://github.com/PolicyEngine/policyengine-us/blob/master/policyengine_us/variables/gov/states/md/tax/income/local/md_local_income_tax_before_credits.py

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
tests failing

**Comment by PavelMakarchuk** [comment]
> tests failing

fixed

File: policyengine_us/variables/gov/states/in/tax/income/in_county_tax.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    reference = "http://iga.in.gov/legislative/laws/2021/ic/titles/006#6-3-2-1"  # (a)(3)
+    defined_for = StateCode.IN
+
+    def formula(tax_unit, period, parameters):
+        # County calculations are at the person level for each taxpayer in the law
+        in_in = tax_unit.household("state_code_str", period) == "IN"
+        county = tax_unit.household("county_str", period)
+        rate = np.zeros_like(county, dtype=float)
+        rates = parameters(period).gov.states["in"].tax.income.county_rates
+        if in_in.sum() > 0:
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/in_county_tax.py:19]: do we need this given we have defined_for?

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/in/tax/income/in_county_tax.py]: we do not, removed

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**